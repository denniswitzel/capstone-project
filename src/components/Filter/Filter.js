import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'


Filter.propTypes = {
    active: PropTypes.string,
    filters: PropTypes.array
}

export default function Filter( {filters, active, setActive }) {
    

    return (
    <ButtonWrapper >
    {filters?.map((filter) => (
    <FilterButton key={filter} active={active === filter} onClick={() => activeFilter(filter)}>{filter.includes('all') ? filter : 'Hide ' + filter}</FilterButton>))}
    </ButtonWrapper>
    )

    function activeFilter(filter){
        setActive(filter)
    }
}


const ButtonWrapper = styled.section`
    display: flex;
    margin-bottom: 20px;
    justify-content: center;
`

const FilterButton = styled.button`
    height: 35px;
    width: 25vw;
    border: 2px solid var(--green);
    color: ${({active}) => active ? 'var(--white)' : 'var(--green)'};
    border-radius: 20px;
    margin: 20px 10px 0 10px;
    background: ${({active}) => active ? 'var(--green)' : 'var(--lightgrey)'};
`