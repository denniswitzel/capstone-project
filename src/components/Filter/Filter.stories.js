import React  from 'react';
import Filter from './Filter';
import { MemoryRouter } from 'react-router-dom';


export default {
  title: 'Components/Filter',
  component: Filter,
  active: 'Show all',
  argTypes: {
    active: { control: {
      type: 'select',
      options: ['Show all', 'gluten', 'soy'],
    }}
  },
};

const filters = ['Show all', 'gluten', 'soy']

const Template = (args) => (
<MemoryRouter>
  <Filter {...args} filters={filters}/>
</MemoryRouter>) 

export const showAll = Template.bind({})
showAll.args = {
  active: 'Show all'
}

export const hideGluten = Template.bind({});
hideGluten.args = {
  active: 'gluten'
}

export const hideSoy = Template.bind({});
hideSoy.args = {
  active: 'soy'
}