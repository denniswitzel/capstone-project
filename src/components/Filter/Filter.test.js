import React from 'react'
import renderer from 'react-test-renderer'
import Filter from './Filter'
import { render  } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { MemoryRouter } from 'react-router-dom'

describe('Filter', () => {
    const filters = ['Show all', 'gluten', 'soy']
    const active = 'Show all'
  it('display filters and active state', () => {
    const { getByText } = render(
        <MemoryRouter>
        <Filter filters={filters} active={active} />
        </MemoryRouter>
)
    expect(['Show all', 'gluten', 'soy']).toEqual(expect.arrayContaining(filters))
    expect(getByText(active)).toBeInTheDocument()
  })

  it('renders correctly', () => {
    const tree = renderer.create(
        <MemoryRouter>
        <Filter/>
        </MemoryRouter>)
    expect(tree).toMatchSnapshot()
  })
})
