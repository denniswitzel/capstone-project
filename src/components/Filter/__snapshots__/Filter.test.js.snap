// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<section
  className="sc-AxjAm ccqfHO"
/>
`;
